src/app/components/cohort-selection/cohort-selection.controller.js

```diff
@@ -1084,16 +1152,30 @@ angular.module('transmartBaseUi')
                                 filterWords: chart.filters()
                             });
                         }
-                        vm.addNodeToActiveCohortSelection(node, filters);
-                    });
+                        var promise = vm.addNodeToActiveCohortSelection(node, filters);
+                        promise.then(function () {
```

Comment:
Is this meant to execute the promises in sequence? It would be useful to add some comments as to what is happening here.